### PR TITLE
.travis.yml: Add Coverity addon

### PR DESCRIPTION
**Pull request by vchong**
Signed-off-by: Victor Chong <victor.chong@linaro.org>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -213,3 +218,17 @@ script:
 
   # Run regression tests (xtest in QEMU)
   - (cd ${HOME}/optee_repo/build && $make check CROSS_COMPILE="ccache arm-linux-gnueabihf-" AARCH32_CROSS_COMPILE=arm-linux-gnueabihf- CFG_TEE_CORE_DEBUG=y DUMP_LOGS_ON_ERROR=1)
+
+env:
+  global:
+    secure: "k30FG3hL6dQywtvMeEjSM0FQ7A0r+AQriUkGSFv7bmwD3EX2H17tJsfZt/h3dj+Pj6lQPdhVg3YO6O6pksJLqBLyz86bN15JEik0e1zftZplzK00oT89eoQr+qgqyrbs6CZBRvS98EAneCNbvgJto9sIPJs0SGu0u31Pr06bTAg="
```
Comment (jbech-linaro) [inline .travis.yml]: Is this a good or a bad thing to have publicly available?

**Comment by vchong** [comment]
Sorry. Was actually in the middle of writing a comment about what you asked and got sidetracked.

This PR just prepares Travis to do a scan if triggered. The trigger is a push to the `coverity_scan` branch, which doesn't yet exist, so nothing will happen for a) and b). c) should not happen since that will just complicate maintenance. The expected work flow is that we only push from `master` to `coverity_scan` when we want a scan to happen, but when to push - that is still under investigation.

File: .travis.yml (hunk 1 above)
Comment (vchong) [inline .travis.yml]: This is ok. All .travis.yml that use this addon have it, e.g. [mbedtls](https://github.com/ARMmbed/mbedtls/blob/development/.travis.yml#L26).

It's not some sort of key but rather an encrypted blob of the Coverity scan token required to upload results to Coverity's server.

**Comment by vchong** [comment]
To clarify earlier comment, nothing will happen for a) and b) as far as scanning is concerned. The normal travis build and test will be performed as they are before this patch. There's no change to that.

**Comment by jbech-linaro** [comment]
Alright, thanks for the explanation, Besides my comment about the email alias, I'm OK with this.
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

File: .travis.yml [hunk 2]
```diff
@@ -213,3 +218,17 @@ script:
...
+
+addons:
+  coverity_scan:
+    project:
+      name: "OP-TEE/optee_os"
+      description: "Build submitted via Travis CI"
+    notification_email: team-security-wg@linaro.org
```
Comment (jforissier) [inline .travis.yml]: +1

**Comment by jforissier** [comment]
LGTM but please add a description of what this does in the commit description.
With this and the email address updated: `Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

File: .travis.yml (hunk 2 above)
Comment (vchong) [inline .travis.yml]: ok

**Comment by vchong** [comment]
Comment addressed and tags added. Ok for merge?

It might also be worth considering doing an actual (manual) push right now, just one time, to (create) the `coverity_scan` branch, just to verify that everything's working as expected, and also to have an initial defects list.

**Comment by jbech-linaro** [comment]
@vchong , you should be able to push the branch (I've granted you the permission to do so). Please feel free to try it out.

**Comment by vchong** [comment]
@jbech-linaro Thanks! The scan was successful so we have an initial list. Now just waiting for Coverity to give approval for viewing it.

**Comment by vchong** [comment]
List is now open for business.

**Pull request merged**